src/middleware/validator.js

```diff
@@ -115,24 +95,19 @@ const validateInput = (ctx, schema) => {
   }
 }
 
-exports.validationMiddleware = directory => async (ctx, next) => {
-  if (directory) {
-    ctx.directory = directory
-    const schema = createValidationSchema(ctx)
-
-    validateInput(ctx, schema)
-    if (!ctx.input) {
-      logger.error(`Validation Failed: ${ctx.body.message}`)
-      return new Error(`Validation Failed: ${ctx.body}`)
-    }
-  } else {
-    logger.error('No input resource name provided')
-    ctx.error = 'Input resource name not given'
+exports.validateInput = validationMap => async (ctx, next) => {
+  try {
+    const schema = createValidationSchema(validationMap)
+    performValidation(ctx, schema)
+  } catch (error) {
+    ctx.error = error
+    return logger.error(`Validation Failed: ${ctx.body.message}`)
```

Comment:
Is the error being thrown different to what is populated in the `ctx.body.message`? maybe we can just log out the error being thrown instead of the `ctx.body.message`

Reply:
I reworked this to check both calls separately and only return the http response and body